src/networking.c

```diff
@@ -405,6 +405,32 @@ void addReplyErrorLength(client *c, const char *s, size_t len) {
 
 /* Do some actions after an error reply was sent (Log if needed, updates stats, etc.) */
 void afterErrorReply(client *c, const char *s, size_t len) {
+    /* Update per command error stats.
+     * If the command was rejected no need to flag it
+     * as a failed count also. */
+    if (c->cmd) {
+        if (c->cmd->flags & CMD_ERR_REJECTED) {
+            c->cmd->rejected_calls++;
+        } else {
+            c->cmd->failed_calls++;
+        }
+    }
```

Comment:
i think this function should only update the error stats, and not the command stats.
in theory a single command can reply with an array of errors (more than one error per call).

i think the right place to update the command stats, is at the end of `call()` and in `rejectCommand()`, both are nontenured to be executed only once per command execution.

this means that the new `CMD_ERR_REJECTED` flag you added isn't needed.
and instead you do need to be able to tell in `call()` that the command failed.
the way to do that can be to try to match a global error counter (much like it looks at changes in `server.dirty`). this new global error counter can also be useful as a general metric (put next to `stat_unexpected_error_replies`)

the only problem with this is nested command (Lua, Multi-Exec, and Modules).
we can probably solve it by adding (yet another server struct global) which we reset to the last global error counter, every time we "log" the error into some command.

Reply:
@oranagra @madolson I've followed the review suggestion and removed the added flag CMD_ERR_REJECTED and moved all command error log to processCommand+rejectCommand. Oran I'm checking within processCommand and not call so that we can check the initial error count and then after call or queing the command we can see if any error reply occurred. Given that rejectCommand/rejectCommandFormat return prior than call this will simplifies stuff and allows to differentiate rejected from failed. 
what do you guys think of this approach?